include/boost/numeric/ublas/tensor/detail/type_traits_tensor.hpp

```diff
@@ -15,152 +15,80 @@
-    struct is_static< dynamic_tensor<T, F> > : std::false_type{};
-
-    template<typename T, typename F>
-    struct is_static_rank< dynamic_tensor<T, F> > : std::false_type{};
-
-    template<typename T, typename F>
-    struct is_dynamic< dynamic_tensor<T, F> > : std::true_type{};
-
-    template<typename T, typename F>
-    struct is_dynamic_rank< dynamic_tensor<T, F> > : std::true_type{};
-
-    template<typename V, typename F>
-    struct result_tensor< V, dynamic_extents<>, F >{
-        using type = dynamic_tensor< V, F >;
-    };
-
-} // namespace boost::numeric::ublas::detail
-
-
-namespace boost::numeric::ublas{
-    
-    template<typename T, std::size_t R, typename F, typename NewValue>
-    struct tensor_rebind< fixed_rank_tensor<T, R, F>, NewValue > {
-        using type = fixed_rank_tensor< NewValue, R, F >;
-    };
+    namespace detail{
+
+        template<typename A, std::size_t N, typename T>
+        struct rebind_static_storage_helper;
+
+        template<typename A, std::size_t N>
+        struct rebind_static_storage_helper<A,N,storage_static_container_tag>
+        {
+            using type = typename storage_traits<A>::template rebind_size<N>;
+        };
+        
+        template<typename A, std::size_t N>
+        struct rebind_static_storage_helper<A,N,storage_resizable_container_tag>
+        {
+            using type = A;
+        };
+
+        template<typename A, std::size_t N>
+        using rebind_static_storage_helper_t = typename rebind_static_storage_helper<A,N, typename storage_traits<A>::resizable_tag >::type;
+        
+    } // namespace detail
     
-    template<typename T, std::size_t R, typename F, typename NewValue, typename NewExtents>
-    struct tensor_rebind< fixed_rank_tensor<T, R, F>, NewValue, NewExtents > {
-        using type = fixed_rank_tensor< NewValue, NewExtents::_size, F >;
-    };
-    
-    template<typename T, std::size_t R, typename F, typename NewValue, typename NewExtents, typename NewLayout>
-    struct tensor_rebind< fixed_rank_tensor<T, R, F>, NewValue, NewExtents, NewLayout > {
-        using type = fixed_rank_tensor< NewValue, NewExtents::_size, NewLayout >;
+    template<typename ValueType, typename A, typename T, T... Ns>
+    struct rebind_storage< basic_static_extents<T,Ns...>, A, ValueType >
+        : std::conditional<
+            std::is_same_v<
+                typename storage_traits<A>::resizable_tag, 
+                storage_static_container_tag
+            >,
+            detail::rebind_static_storage_helper_t<
+                typename storage_traits<A>::template rebind<ValueType>,
+                ( ... * Ns)
+            >,
+            typename storage_traits<A>::template rebind<ValueType>
+        >
+    {};
+
+    template<typename ValueType, typename A, typename T>
+    struct rebind_storage< basic_static_extents<T>, A, ValueType >
+    {
+        using type = typename storage_traits<A>::template rebind<ValueType>;
     };
```

Comment:
Why do we need this.

Reply:
This is a standard way to get the storage type because the storage may not be `std::array` or `std::vector`.

Reply:
Why not defining the typical storage_type_traits for container types ?

Reply:
This will introduce too many variables and `storage_traits` already exists. I can move this to new header `storage_type_traits`.

Reply:
I like this trait, except for its location. Amit did a great job at making sure that all containers run fine. The most useful use case I can come up with is that we can use `CUDA Thrust` Vectors easily ( `host_vector` & `device_vector`). See [this](https://docs.nvidia.com/cuda/thrust/index.html) . If we restrict to standard types, we will again need it someday when adding tensor to work on GPUs.

Reply:
I was talking about `rebing_storage`